test/test_SW_Flow_Lib.cc

```diff
@@ -308,7 +308,1074 @@ TEST(SWFlowTest, LitterInterceptedWater) {
+  TEST(SWFlowTest, evap_fromSurface){
+
+    //INPUTS
+    double water_pool = 1, evap_rate = 0.33, aet = 0.53;
+
+    //Begin TEST for when water_pool > evap_rate
+    double aetExpected = 0.86, evapExpected = 0.33, waterExpected = 0.67;
+    evap_fromSurface(&water_pool, &evap_rate, &aet);
+
+    EXPECT_NEAR(aet, aetExpected, 0.0000001); //Variable aet is expected to be 0.86 with current inputs
+    EXPECT_GE(aet, 0); //aet is never negative
+
+    EXPECT_NEAR(evap_rate, evapExpected, 0.0000001); //Variable evap_rate is expected to be 0.33 with current inputs
+    EXPECT_GE(evap_rate, 0); //evap_rate is never negative
+
+    EXPECT_NEAR(water_pool, waterExpected, 0.0000001); //Variable water_pool is expected to be 0.67 with current inputs
+    EXPECT_GE(water_pool, 0); //water_pool is never negative
+
+    //Begin TEST for when water_pool < evap_rate
+    water_pool = 0.1, evap_rate = 0.67, aet = 0.78, aetExpected = 0.88, evapExpected = 0.1;
+    evap_fromSurface(&water_pool, &evap_rate, &aet);
+
+    EXPECT_NEAR(aet, aetExpected, 0.0000001); //Variable aet is expected to be 0.88 with current inputs
+    EXPECT_GE(aet, 0); //aet is never negative
+
+    EXPECT_NEAR(evap_rate, evapExpected, 0.0000001); //Variable evap_rate is expected to be 0.1 with current inputs
+    EXPECT_GE(evap_rate, 0); //evap_rate is never negative
+
+    EXPECT_DOUBLE_EQ(water_pool, 0); //Variable water_pool is expected to be 0 when water_pool < evap_rate
+    EXPECT_GE(water_pool, 0); //water_pool is never negative
+
+    //Reset to previous global states.
+    Reset_SOILWAT2_after_UnitTest();
+  }
+
+  //Test remove_from_soil when nlyrs = 1 and when nlyrs = MAX
+  TEST(SWFlowTest, remove_from_soil){
+
+    //INPUTS
+    double swc[25] =  {0.11, 0.12, 0.13, 0.14, 1.91, 1.92, 1.93, 3.81, 3.82, 3.83, 5.71, 5.72, 5.73, 7.61, 7.62,
+        7.63, 9.51, 9.52, 9.53, 11.41, 11.42, 11.43, 13.31, 13.32, 13.33};
+    double qty[25] = {0.01, 0.02, 0.03, 0.04, 0.05, 1.51, 3.51, 5.51, 7.51, 9.51, 11.51, 13.51, 0.01, 0.02, 0.03,
+        0.04, 0.05, 1.51, 3.51, 5.51, 7.51, 9.51, 11.51, 13.51, 15};
+    double aet = 0.33, rate = 0.62;
+    unsigned int nlyrs = 25;
+    double coeff[25] = {0.0496, 0.0495, 0.1006, 0.1006, 0.1006, 0.1997, 0.1997, 0.1997, 0.0496, 0.0495, 0.1006,
+        0.1006, 0.1006, 0.1997, 0.1997, 0.1997, 0.0496, 0.0495, 0.1006, 0.1006, 0.1006, 0.1997, 0.1997, 0.1997, 0.1999};
+    double coeffZero[25] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
+    double swcmin[25] = {0.01, 0.02, 0.03, 0.04, 1.71, 1.72, 1.73, 3.61, 3.62, 3.63, 5.51, 5.52, 5.53, 7.41, 7.42,
+        7.43, 9.31, 9.32, 9.33, 11.11, 11.12, 11.13, 13.01, 13.02, 13.03};
+
+    //Begin TEST when nlyrs = MAX LAYERS
+    //TEST for if local variable sumswp = 0 (coeff[i] = 0)
+
+    //INPUTS for expected outputs
+    double swcExpected[25] =  {0.11, 0.12, 0.13, 0.14, 1.91, 1.92, 1.93, 3.81, 3.82, 3.83, 5.71, 5.72, 5.73,
```

Comment:
it is not clear that this ought to be identical values to `swc`, `qty`, etc.

--> write code that makes this explicitly clear: e.g., 
```
ForEachSoilLayer(i) {
  swcExpected[i] = swc[i];
  ...
}
```